From d481b87a4249cb405e91cd43fba72b2588670580 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 15 Jul 2021 06:38:43 +0300
Subject: [PATCH] [SW-44497] preboot: refactor thermal_v2 for gaudi2

Currently thermal_v2 is compiled only for greco, which is to be
supported for gaudi2 also.

This patch removes greco specific header files currently in thermal_v2
and moves hl_ts_reset() to platform specific code as it is not common
between greco and gaudi2.

This is in preparation to support ts functionality in gaudi2.

Change-Id: I0b70dad29c7eaf95741c386a1ef72b265e54531c
---

diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index 599b1bc..71ac1a0 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -5,4 +5,27 @@
  *
  */
 
-/* TODO: Commons functions in arc, arc_el32, arm to be moved here. */
+#include <psoc_global_conf_regs.h>
+#include <preboot.h>
+#include <pll.h>
+#include <uart.h>
+
+#define UNIT_RST_N_L_MSK		0xF0000
+#define UNIT_RST_N_M_MSK		0x0
+#define UNIT_RST_N_H_MSK		0x0
+
+void hl_ts_reset(void)
+{
+	preboot_puts("Performing RST of TS UNIT\n");
+	/* Perform unit address */
+	regmap_writel(0, mmPSOC_GLOBAL_CONF_UNIT_RST_N);
+	regmap_writel(UNIT_RST_N_L_MSK, mmPSOC_GLOBAL_CONF_UNIT_RST_N_L);
+	regmap_writel(0, mmPSOC_GLOBAL_CONF_UNIT_RST_N_M);
+	regmap_writel(0, mmPSOC_GLOBAL_CONF_UNIT_RST_N_H);
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_UNIT_RST_N);
+
+	udelay(1000);
+
+	regmap_writel(0, mmPSOC_GLOBAL_CONF_UNIT_RST_N_L);
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_UNIT_RST_N);
+}
diff --git a/lib/preboot/thermal_v2.c b/lib/preboot/thermal_v2.c
index 2069b62..091ae8a 100644
--- a/lib/preboot/thermal_v2.c
+++ b/lib/preboot/thermal_v2.c
@@ -5,11 +5,7 @@
  *
  */
 
-#include <greco/greco_reg_map.h>
-#include <goya2_blocks_linux_driver.h>
 #include <psoc_global_conf_regs.h>
-#include <dcon0_tstd_masks.h>
-#include <dcon0_tstd_regs.h>
 
 #include <preboot.h>
 #include <pll.h>
@@ -50,10 +46,6 @@
 #define HL_PVT_TM_SCRATCH_PATTERN		0xAAAAAAAA
 #define HL_TS_PVT_INIT_FAILED			0x1
 #define HL_TS_IP_CTRL_SHUTDOWN			1
-
-#define UNIT_RST_N_L_MSK			0xF0000
-#define UNIT_RST_N_M_MSK			0x0
-#define UNIT_RST_N_H_MSK			0x0
 
 void hl_set_calib_coeff(int ts, bool calib_enabled,
 		struct hl_tmp_opmode_info *calib_data)
@@ -251,18 +243,4 @@
 	else
 		preboot_puts_hex("Failed to disable TS data acquisitions for ",
 					ts, false);
-}
-
-void hl_ts_reset(void)
-{
-	preboot_puts("Performing RST of TS UNIT\n");
-	/* Perform unit address */
-	regmap_writel(0, mmPSOC_GLOBAL_CONF_UNIT_RST_N);
-	regmap_writel(UNIT_RST_N_L_MSK, mmPSOC_GLOBAL_CONF_UNIT_RST_N_L);
-	regmap_writel(0, mmPSOC_GLOBAL_CONF_UNIT_RST_N_M);
-	regmap_writel(0, mmPSOC_GLOBAL_CONF_UNIT_RST_N_H);
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_UNIT_RST_N);
-	udelay(1000);
-	regmap_writel(0, mmPSOC_GLOBAL_CONF_UNIT_RST_N_L);
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_UNIT_RST_N);
 }
